src/components/Dialogs/Conditions/AlphanumericCondition.js

```diff
@@ -0,0 +1,171 @@
+export const OPERATOR_EMPTY = `EQ:${NULL_VALUE}`
+export const OPERATOR_NOT_EMPTY = `NE:${NULL_VALUE}`
+export const OPERATOR_EQUAL_CASE_SENSITIVE = 'IEQ'
+export const OPERATOR_NOT_EQUAL_CASE_SENSITIVE = '!IEQ'
+export const OPERATOR_CONTAINS_CASE_SENSITIVE = 'ILIKE'
+export const OPERATOR_NOT_CONTAINS_CASE_SENSITIVE = '!ILIKE'
+
+const operators = {
+    [OPERATOR_EQUAL]: i18n.t('exactly'),
+    [OPERATOR_NOT_EQUAL]: i18n.t('is not'),
+    [OPERATOR_CONTAINS]: i18n.t('contains'),
+    [OPERATOR_NOT_CONTAINS]: i18n.t('does not contain'),
+    [OPERATOR_EMPTY]: i18n.t('is empty / null'),
+    [OPERATOR_NOT_EMPTY]: i18n.t('is not empty / not null'),
+}
+
+const AlphanumericCondition = ({
+    condition,
+    onChange,
+    onRemove,
+    allowCaseSensitive,
+}) => {
+    let operator, value
+
+    if (condition.includes(NULL_VALUE)) {
+        operator = condition
+    } else {
+        const parts = condition.split(':')
+        operator = parts[0]
+        value = parts[1]
+    }
+
+    const setOperator = (input) => {
+        if (input.includes(NULL_VALUE)) {
+            onChange(`${input}`)
+        } else {
+            onChange(`${input}:${value || ''}`)
+        }
+    }
+
+    const setValue = (input) => {
+        onChange(`${operator}:${input || ''}`)
+    }
+
+    const toggleCaseSensitive = (state) => {
+        switch (operator) {
+            case OPERATOR_EQUAL:
+            case OPERATOR_EQUAL_CASE_SENSITIVE:
+                setOperator(
+                    state ? OPERATOR_EQUAL_CASE_SENSITIVE : OPERATOR_EQUAL
+                )
+                break
+            case OPERATOR_NOT_EQUAL:
+            case OPERATOR_NOT_EQUAL_CASE_SENSITIVE:
+                setOperator(
+                    state
+                        ? OPERATOR_NOT_EQUAL_CASE_SENSITIVE
+                        : OPERATOR_NOT_EQUAL
+                )
+                break
+            case OPERATOR_CONTAINS:
+            case OPERATOR_CONTAINS_CASE_SENSITIVE:
+                setOperator(
+                    state ? OPERATOR_CONTAINS_CASE_SENSITIVE : OPERATOR_CONTAINS
+                )
+                break
+            case OPERATOR_NOT_CONTAINS:
+            case OPERATOR_NOT_CONTAINS_CASE_SENSITIVE:
+                setOperator(
+                    state
+                        ? OPERATOR_NOT_CONTAINS_CASE_SENSITIVE
+                        : OPERATOR_NOT_CONTAINS
+                )
+                break
+        }
+    }
+
+    const useGenericOperator = (operator) => {
+        switch (operator) {
+            case OPERATOR_EQUAL_CASE_SENSITIVE:
+                return OPERATOR_EQUAL
+            case OPERATOR_NOT_EQUAL_CASE_SENSITIVE:
+                return OPERATOR_NOT_EQUAL
+            case OPERATOR_CONTAINS_CASE_SENSITIVE:
+                return OPERATOR_CONTAINS
+            case OPERATOR_NOT_CONTAINS_CASE_SENSITIVE:
+                return OPERATOR_NOT_CONTAINS
+            default:
+                return operator
+        }
+    }
```

Comment:
I think this would work too. If the case sensitive suffix is not present, nothing is replaced.
```suggestion
    const useGenericOperator = (operator) => operator.replace(suffix, '')
```